lib/api.js

```diff
@@ -1474,19 +1433,28 @@ function getFormattedTraceId (options = {}) {
  *
  * logger.info(ao.insertLogObject({
  *     message: 'this object is being modified by insertLogObject',
- *     more: 'there will be an added ao property'
+ *     more: 'there will be an added sw property'
  * }))
  */
 function insertLogObject (o = {}) {
   // if truthy and tracing insert it based on sample setting. otherwise if 'always'
   // then insert a trace ID regardless. No explicit check for 'traced' is required.
   let last
-  if (ao.cfg.insertTraceIdsIntoLogs && (last = ao.lastEvent)) {
+  if ((ao.cfg.insertTraceIdsIntoLogs || ao.cfg.insertTraceIdsIntoMorgan) && (last = ao.lastEvent)) {
     if (ao.cfg.insertTraceIdsIntoLogs !== 'sampledOnly' || last.event.getSampleFlag()) {
```

Comment:
should `|| ao.cfg.insertTraceIdsIntoMorgan !== 'sampledOnly'` be added to this condition as well?

Reply:
Sharp eyes...

The answer to the one question is of the "**yes, but no, then maybe, so I don't know**" category.

**Yes**, adding the condition would cover all the cases and make the code clear, **but no**, because that code would be redundant as the sampledOnly case is covered in the probe returning an empty string before calling the api method above, **then maybe** the probe code could be changed to not do that, or both the api code and probe code could be changed to no have a probe specific config conditional in the general API, or maybe the probe specific global config setting could be dropped all together, or maybe... **so I don't know**... 

Ask was to trace context in logs... We'll (maybe) deal with the API/Config vars later

Reply:
Do you have test coverage in case the probe code changes?

Reply:
If the [probe conditional](https://github.com/appoptics/appoptics-apm-node/blob/cd6f461af51f9e62a0d40df909d1d60f02ef0d0f/lib/probes/morgan.js#L50) is removed the then a test from [this test array](https://github.com/appoptics/appoptics-apm-node/blob/cd6f461af51f9e62a0d40df909d1d60f02ef0d0f/test/probes/morgan.test.js#L244) will fail. But, note caveat re: morgan tests in PR description and the general complexity of code/test interfaces...